icaruscode/Timing/PMTWaveformTimeCorrectionExtractor.cxx

```diff
@@ -47,11 +47,11 @@ icarus::timing::PMTWaveformTimeCorrectionExtractor::Error::Error
 
 icarus::timing::PMTWaveformTimeCorrectionExtractor::MultipleCorrectionsForChannel
 ::MultipleCorrectionsForChannel
-  (unsigned int existing, unsigned int additional)
+  (unsigned int channel, unsigned int existing, unsigned int additional)
     : Error{
-      "Attempt to overwrite correction from channel "
-      + std::to_string(existing) + " with one from channel "
-      + std::to_string(additional) + "\n"
+      "Attempt to overwrite the correction for channel "
+      + std::to_string(channel) + " (from channel " + std::to_string(existing)
+      + ") with another (from channel " + std::to_string(additional) + ")\n"
```

Comment:
I'm a little confused by these changes. What is this exception trying to catch?

Reply:
This is about the timing correction from the digitised waveforms (the reference probably will always be the global trigger signal), which we use to synchronise the waveforms from all the readout boards. Cable delay corrections are also added at the same time.
The action starts from the decoder, which after decoding and appropriately merging all the waveforms is now extracting time corrections. Corrections may be extracted from all the special waveforms (the ones on the 16th channel). The decoder is filling a table (`vector`) with a correction value for each regular PMT channel, by asking  the correction extractor to process each of the special waveforms ([`icarus::timing::PMTWaveformTimeCorrectionExtractor::findWaveformTimeCorrections`](icaruscode/Timing/PMTWaveformTimeCorrectionExtractor.cxx#L174)). That function extracts the correction value from one special waveform (an unassuming [`getStartSample()`](icaruscode/Timing/PMTWaveformTimeCorrectionExtractor.cxx#L195) and then fills the correction value for all the (45) regular channels in the crate that waveform belongs to.
This exception is thrown when this happens while one of the waveforms has already a correction, which is not expected. For example, _imagine_, if you can, that the mapping is wrong and a channel ID, say, `225`, appears multiple times, either in the same or in a different one. That means that the channel is registered in two different crates (or in the same crate twice). When we process the special waveform for the first crate, we assign all its 45 channels a correction value (e.g. from channel `0x6051`), including channel `225`. Then, when we meet the special waveform of the second crate (`0x6061`?), we do the same, 45 channels, and `225` again among them. There is where the problem shows. If the two instances appear under the same crate, little changes (it would show twice `0x6051` instead of mentioning `0x6051` and `0x6061`).
The message is now telling which special channel the already `existing` correction was extracted from, from which special channel we are trying to extract it again, and the regular channel we were filling the correction for.

Reply:
Thank you! it's clear now.